nyc/bills.py

```diff
@@ -161,133 +161,150 @@ def sponsorships(self, matter_id):
+                              action['action_date'],
+                              organization={'name': action['responsible_org']},
+                              classification=action['classification'])
 
+        result, votes = vote
 
-            if text:
-                if text['MatterTextPlain']:
-                    bill.extras['plain_text'] = text['MatterTextPlain'].replace(u'\u0000', '')
+        if result:
+            organization = json.loads(act['organization_id'].lstrip('~'))
+            vote_event = VoteEvent(legislative_session=bill.legislative_session,
+                                   motion_text=act['description'],
+                                   organization=organization,
+                                   classification=None,
+                                   start_date=act['date'],
+                                   result=result,
+                                   bill=bill)
 
-                if text['MatterTextRtf']:
-                    bill.extras['rtf_text'] = text['MatterTextRtf'].replace(u'\u0000', '')
+            legistar_web, legistar_api = [src['url'] for src in bill.sources]
 
+            vote_event.add_source(legistar_web)
+            vote_event.add_source(legistar_api + '/histories')
 
-            for action, vote in self.actions(matter_id):
-                act = bill.add_action(action['action_description'],
-                                      action['action_date'],
-                                      organization={'name': action['responsible_org']},
-                                      classification=action['classification'])
+            for vote in votes:
+                raw_option = vote['VoteValueName'].lower()
 
-                result, votes = vote
+                if raw_option == 'suspended':
+                    return
 
-                if result:
-                    organization = json.loads(act['organization_id'].lstrip('~'))
-                    vote_event = VoteEvent(legislative_session=bill.legislative_session,
-                                           motion_text=act['description'],
-                                           organization=organization,
-                                           classification=None,
-                                           start_date=act['date'],
-                                           result=result,
-                                           bill=bill)
+                clean_option = self.VOTE_OPTIONS.get(raw_option, raw_option)
+                vote_event.vote(clean_option, vote['VotePersonName'].strip())
 
-                    vote_event.add_source(legistar_web)
-                    vote_event.add_source(legistar_api + '/histories')
+            return vote_event
 
-                    for vote in votes:
-                        raw_option = vote['VoteValueName'].lower()
 
-                        if raw_option == 'suspended':
-                            continue
+    def scrape(self, window=3, matter_id=None):
```

Comment:
Do we want to be able to pass in more than one matter_id in at a time?

Reply:
That's a good question. To me, `fetch` implies you're getting one thing. Does it have a similar connotation to you?

Reply:
Ah, scratch that, I guess there's nothing stopping us from calling `fetch` multiple times downstream, eh?

Reply:
Bearing that in mind, it'd be great to support more than one ID. Good idea.